pkg/flare/flare.go

```diff
@@ -31,6 +31,11 @@ type flareResponse struct {
 
 // SendFlareWithHostname sends a flare with a set hostname
 func SendFlareWithHostname(archivePath string, caseID string, email string, hostname string) (string, error) {
+
+	var redirectHops = 0
+	var url = mkURL(caseID)
+
+redirect:
```

Comment:
With the goto change, you're spawning a routine and creating a new pipe (reader/writer) for each hop, is this really what you want?

Reply:
Why is it not enough to just re-do the requests (line 80)?

Reply:
Because the pipe is already "read" and closed. Re-creating the entire thing might be a bit of an overkill, but it's the solution that requires less code changes.